crates/bevy_math/src/smooth_damp.rs

```diff
@@ -0,0 +1,170 @@
+use crate::{Vec2, Vec3};
+
+pub struct SmoothDamp<T> {
+    pub position: T,
+    pub velocity: T,
+}
+
+pub trait SmoothDampFunctions {
+    /// Smooths value to a goal using a damped spring.
+    ///
+    /// `smooth_time` is the expected time to reach the target when at maximum velocity.
+    ///
+    /// Returns smoothed value and new velocity.
+    ///
+    /// # Example
+    /// ```
+    /// # use bevy_math::prelude::{Vec3, Quat};
+    /// # use bevy_math::{SmoothDamp, SmoothDampFunctions};
+    /// # struct Transform {
+    /// #     translation: Vec3,
+    /// #     rotation: Quat,
+    /// #     scale: Vec3
+    /// # }
+    /// struct SmoothTransform {
+    ///     pub smoothness: f32,
+    ///     pub target: Vec3,   
+    ///     velocity: Vec3   
+    /// }
+    ///
+    /// fn smooth_transform_update(dt: f32, transform: &mut Transform, smoother: &mut SmoothTransform) {
+    ///     let SmoothDamp { position, velocity } = Vec3::smooth_damp(
+    ///         transform.translation,
+    ///         smoother.target,
+    ///         smoother.velocity,
+    ///         smoother.smoothness,
+    ///         dt,
+    ///     );
+    ///     transform.translation = position;
+    ///     smoother.velocity = velocity;
+    /// }
+    /// ```
+    fn smooth_damp(
+        from: Self,
+        to: Self,
+        velocity: Self,
+        smooth_time: f32,
+        delta_time: f32,
```

Comment:
I'm curious if we would want to make these generic?
e.g. would it be viable for a user to want to use a `f64` or some other floating/fixed-point type?
In which case this could become a associated type.